resource-providers/src/main/java/io/opentelemetry/resource/spring/SpringBootServiceNameGuesser.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.resource.spring;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.ResourceProvider;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.yaml.snakeyaml.Yaml;
+
+/**
+ * A ResourceProvider that will attempt to guess the application name for a Spring Boot service.
+ * When successful, it will return a Resource that has the service name attribute populated with the
+ * name of the Spring Boot application. It uses the following strategies, and the first successful
+ * strategy wins:
+ *
+ * <ul>
+ *   <li>Check for the SPRING_APPLICATION_NAME environment variable
+ *   <li>Check for spring.application.name system property
+ *   <li>Check for application.properties file on the classpath
+ *   <li>Check for application.properties in the current working dir
+ *   <li>Check for application.yml on the classpath
+ *   <li>Check for application.yml in the current working dir
+ *   <li>Check for --spring.application.name program argument (not jvm arg) via ProcessHandle
+ *   <li>Check for --spring.application.name program argument via sun.java.command system property
+ * </ul>
+ */
+@AutoService(ResourceProvider.class)
+public class SpringBootServiceNameGuesser implements ResourceProvider {
+
+  private static final Pattern COMMANDLINE_PATTERN =
+      Pattern.compile(".*--spring\\.application\\.name=([a-zA-Z.\\-_]+).*");
```

Comment:
I wanted to include the hyphen `-` and the period `.` because I think people use those pretty frequently in application names and they are not part of `\w`. 😿

Reply:
you could just take everything up to the first space `([^ ]+)`